### PR TITLE
Remove glassfish dependency platform TCK build

### PR DESCRIPTION
**Pull request by alwin-joseph**
---
name: Pull Request
about: Create a pull request for a Platform TCK change
title: ''
labels: ''
assignees: ''

---

**Fixes Issue**
https://github.com/eclipse-ee4j/jakartaee-tck/issues/583 - addresses the platform TCK build alone.

**Related Issue(s)**
Specify any related issue(s) links.

**Describe the change**
The change downloads api jars from maven that is used to resolve compilation issues for building platform TCK. The change passes the signaturetest and sample test. More testing will follow.

**Additional context**
The jars that are required are mentioned in docker/pom.xml. The dependency on metro jars(org.glassfish.metro) is also under question. Will check more if those can be replaced too.

Also appended the platform TCK version to 9.1.0 

CC @edbratt @scottmarlow @gurunrao

### CODE REVIEW

**Review by RohitKumarJain — APPROVED** [review]
LGTM !!! Maybe we can raise another commit for updating the Copyright year.

File: docker/build_jakartaeetck.sh [hunk 1]
```diff
@@ -75,40 +70,34 @@ ant -version
...
-ls -l $GF_HOME/$GF_TOPLEVEL_DIR/glassfish/
 
+mkdir -p $JAKARTA_JARS/modules
+mkdir -p $JAKARTA_JARS/endorsed
 
-if [ ! -z "$GF_VERSION_URL" ]; then
-  wget --progress=bar:force --no-cache $GF_VERSION_URL -O glassfish.version
-  cat glassfish.version
-fi
+mvn -f $BASEDIR/docker/pom.xml dependency:copy-dependencies -DoutputDirectory="${JAKARTA_JARS}/modules" -Dmdep.stripVersion=true
```
Comment (scottmarlow) [inline docker/build_jakartaeetck.sh:98]: +1000 for using Maven for downloading the SPEC API dependencies which [sort of was brought up by Bill before](https://github.com/eclipse-ee4j/jakartaee-tck/issues/207#issuecomment-617482697).

File: docker/pom.xml [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+-->
+ <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+    <groupId>jakarta</groupId>
+    <version>1</version>
+    <artifactId>dependency_copy</artifactId>
+
+  <dependencies>
```
Comment (scottmarlow) [inline docker/pom.xml:25]: The SPEC API versions are correct, looks good!

File: docker/pom.xml [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    </dependency>
+    <dependency>
+        <groupId>jakarta.xml.bind</groupId>
+        <artifactId>jakarta.xml.bind-api</artifactId>
+        <version>3.0.0</version>
+    </dependency>
+    <dependency>
+        <groupId>com.sun.xml.bind</groupId>
+        <artifactId>jaxb-osgi</artifactId>
+        <version>3.0.0</version>
```
Comment (scottmarlow) [inline docker/pom.xml:169]: Is com.sun.xml.bind:jaxb-osgi a GlassFish dependency that we need here?  The same question for the below dependencies?

File: docker/pom.xml [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+  <dependencies>
+    <dependency>
+        <groupId>jakarta.activation</groupId>
+        <artifactId>jakarta.activation-api</artifactId>
+        <version>2.0.0</version>
+    </dependency>
+    <dependency>
+        <groupId>jakarta.annotation</groupId>
+        <artifactId>jakarta.annotation-api</artifactId>
+        <version>2.0.0</version>
```
[the remark this answers is not in the document]
Comment (alwin-joseph) [reply docker/pom.xml]: Done.

File: docker/build_jakartaeetck.sh [hunk 5]
```diff
@@ -75,40 +70,34 @@ ant -version
...
 cd $BASEDIR
-if [ -z "$GF_BUNDLE_URL" ]; then
-  echo "Using default url for GF bundle: $DEFAULT_GF_BUNDLE_URL"
-  export GF_BUNDLE_URL=$DEFAULT_GF_BUNDLE_URL
-fi
-wget --progress=bar:force --no-cache $GF_BUNDLE_URL -O latest-glassfish.zip
-unzip -q -o latest-glassfish.zip
-ls -l $GF_HOME/$GF_TOPLEVEL_DIR/glassfish/
 
+mkdir -p $JAKARTA_JARS/modules
```
[the remark this answers is not in the document]
Comment (alwin-joseph) [reply docker/build_jakartaeetck.sh:95]: I also created $JAKARTA_JARS/endorsed to be refered in other places hence created modules dir.

File: docker/pom.xml (hunk 3 above)
Comment (alwin-joseph) [reply docker/pom.xml]: I see those jars are required during the build, the exact class files used is not known. I will investigate this to identify the classes for a different change.

**Review by gurunrao — APPROVED** [review]

**Pull request merged**